osconfig_tests/test_suites/package_management/package_management_test_data.go

```diff
@@ -116,29 +139,45 @@ func addPackageRemovalTest(pkgTestSetup []*packageManagementTestSetup) []*packag
 	for _, pkgManager := range pkgManagers {
 		var oc *osconfigpb.OsConfig
 		var assign *osconfigpb.Assignment
-		var instaneName, ss, vs string
+		var instanceName, image, ss, vs string
 
 		switch pkgManager {
 		case "apt":
-			instaneName = fmt.Sprintf("%s-%s", filepath.Base(debianImage), testName)
+			image = debianImage
+			instanceName = fmt.Sprintf("%s-%s", path.Base(debianImage), testName)
 			pkg := osconfigserver.BuildPackage(packageName)
 			pkgs := []*osconfigpb.Package{pkg}
 			oc = osconfigserver.BuildOsConfig(testName, desc, osconfigserver.BuildAptPackageConfig(nil, pkgs, nil), nil, nil, nil, nil)
-			assign = osconfigserver.BuildAssignment(testName, desc, osconfigserver.BuildInstanceFilterExpression(instaneName), []string{fmt.Sprintf("projects/%s/osConfigs/%s", testProjectID, oc.Name)})
+			assign = osconfigserver.BuildAssignment(testName, desc, osconfigserver.BuildInstanceFilterExpression(instanceName), []string{fmt.Sprintf("projects/%s/osConfigs/%s", testProjectID, oc.Name)})
 			ss = `%s
 			sudo apt-get -y install %s;
 			while true; do /usr/bin/dpkg-query -s %s | sudo tee /dev/ttyS0;
 			sleep 5;
 			done`
 			ss = fmt.Sprintf(ss, utils.InstallOSConfigDeb, packageName, packageName)
 			vs = fmt.Sprintf("package '%s' is not installed", packageName)
+		case "yum":
+			image = centosImage
+			instanceName = fmt.Sprintf("%s-%s", path.Base(image), testName)
+			pkg := osconfigserver.BuildPackage(packageName)
+			removePkg := []*osconfigpb.Package{pkg}
+			oc = osconfigserver.BuildOsConfig(testName, desc, nil, osconfigserver.BuildYumPackageConfig(nil, removePkg, nil), nil, nil, nil)
+			assign = osconfigserver.BuildAssignment(testName, desc, osconfigserver.BuildInstanceFilterExpression(instanceName), []string{fmt.Sprintf("projects/%s/osConfigs/%s", testProjectID, oc.Name)})
+			ss = `%s
+			sudo yum -y install %s
```

Comment:
You need to check output for these install calls, if this package did not exist in a repo or the install failed you would get a false positive for the test. Also restart the service before checking if the pacakge is removed otherwise you will be stuck waiting.

Reply:
add check for package installation before moving forward to test assertion